File: TwoPointers/python/11_Container_With_Most_Water.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        left, right = 0, len(height) - 1
        max_area = 0

        while left < right:
            area = (min(height[left], height[right])) * (right - left)
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
            max_area = max(area, max_area)

        return max_area

File: TwoPointers/js/11_Container_With_Most_Water.js
var maxArea = function(height) {
    let left = 0;
  let right = height.length - 1;
  let _maxArea = 0;

  while (left < right) {
    let area = Math.min(height[left], height[right]) * (right - left);

    if (height[left] > height[right]) {
      right--;
    } else {
      left++;
    }
    _maxArea = Math.max(area, _maxArea);
  }

  return _maxArea;
};